pulsar-client/src/main/java/org/apache/pulsar/client/impl/MultiTopicsConsumerImpl.java

```diff
@@ -1045,6 +1054,12 @@ private void doSubscribeTopicPartitions(Schema<T> schema,
                         String partitionName = TopicName.get(topicName).getPartition(partitionIndex).toString();
                         CompletableFuture<Consumer<T>> subFuture = new CompletableFuture<>();
                         configurationData.setStartPaused(paused);
+                        BatchReceivePolicy internalBatchReceivePolicy = BatchReceivePolicy.builder()
+                                .maxNumMessages(Math.max(receiverQueueSize / 2, 1))
+                                .maxNumBytes(-1)
+                                .timeout(0, TimeUnit.MILLISECONDS)
+                                .build();
+                        configurationData.setBatchReceivePolicy(internalBatchReceivePolicy);
```

Comment:
There are other 2 places creating `ConsumerImpl` in this class.

Reply:
@Jason918 Fixed.